support/build/Makefile.rules

```diff
@@ -252,21 +252,30 @@ verbose_cmd = @printf '  %-7s %s\n' '$(1)'  '$(2)' && $(3)
 verbose_cmd_inner = printf '  %-7s %s\n' '$(1)'  '$(2)' && $(3)
 endif
 
+# Writes arguments into a build_args file
+#
+# write_build_args $target,$longargs(optional)
+ifeq ($(CONFIG_BUILD_ARGS_FILE),y)
+write_build_args = $(if $(2),$(file > $(addsuffix .build_args,$(1)),$(2))@$(addsuffix .build_args,$(1)),)
```

Comment:
Do you think it is possible to write down the whole command string (including the command to be executed)? Would this work? If so, I personally would prefer creating a `.cmd` file.

Reply:
That should be possible, I thought about it but did not test it. The limit we run in is the one for [a single argument](https://stackoverflow.com/q/64201512), the limit for the complete argument is much bigger, but there is still a limit.

The problem for fixdep and the second expansion are independent from that.

Reply:
Hum, my goal would be to fix this transparently ehind the scenes with build_cmd plus having a nice way to check later which arguments and command was called to create a file. This could work agnostic to the chosen compiler (so no need to change any buildrule, avoids compatibility problems). If possible, I would keep the prettyfying `echo` command within within the build system but the actual build command in a `.cmd` file. Probably `fixdep` would also go there. Would this fix or avoid the fixdep problem that you mentioned?

Reply:
> my goal would be to fix this transparently behind the scenes with build_cmd (no new argument) plus having a nice way to check later which arguments and command was called to create a file.

Yes, that should work as long as the commands stay in the (much bigger) limit of the overall argument size.

> Would this fix or avoid the fixdep problem that you mentioned?

Unfortunately not.
`fixdep` gets passed the command to figure out if it changes: `$Q $(UK_FIXDEP) $(call tmp_depfile,$3) $3 '$(call strip,$4)' ...` The 3rd parameter is the whole command as one argument, which will cause the original issue to reappear.
Omitting it will make `fixdep` ignore changes to the commandline. I think passing it as hash of the command instead of the actual command should work, but it is really hard to tell.

Reply:
To be clear, I haven't run into an actual problem with `fixdep` that I know of.

On the other hand, this argument is there for a reason, and I don't want to break the build system in subtle ways.

Reply:
Ah, okay, we are not using that part of fixdep yet - so I assume the problem you mention is currently irrelevant. In truth, we still need to find a way to force recompilation if some compiler arguments change (and not the file itself). This parameter of fixdep could have been helpful for this but maybe `.cmd` files would even be the better way for us.
Do you agree to continue with `cmd` file that contai command and arguments?

Reply:
Yes, I will switch to writing the whole command.

Then what is left if the issue with the double expansion. I struggling to to write the command file only in the second expansion.

Reply:
:+1: